### DISCUSSION TITLE
Turn on/off sensors

### DISCUSSION THREAD

**Discussion by simonpwong** [dagster-io/dagster#20823] [Q&A]
I have a very simple schedule that executes a few assets and runs daily at midnight, the whole run takes less than 30 seconds. I have set up a **run_failure_sensor** that triggers if any assets fail and a **run_status_sensor** that triggers if the full run is successful. These sensors send a success/fail message notification to my discord.

Given the schedule completes in under 30 seconds, is there a way to automatically switch the sensors on around the time of each scheduled run and switch off shortly after the scheduled run is complete? If this is considered bad practice, is there any alternative solutions?

**Comment by garethbrickman** [comment]
Sensors in Dagster are designed to continuously monitor for specific conditions and then react accordingly, rather than being toggled on and off around specific events. Using [Run Status Sensors](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-status-sensors), as you are, is best practice!
